paasta_tools/async_utils.py

```diff
@@ -27,8 +27,10 @@ def async_ttl_cache(
 ) -> Callable[
     [Callable[..., Awaitable[T]]], Callable[..., Awaitable[T]]  # wrapped  # inner
 ]:
-    async def call_or_get_from_cache(cache, async_func, args, kwargs):
-        key = functools._make_key(args, kwargs, typed=False)
+    async def call_or_get_from_cache(
+        cache, async_func, args_without_self, args, kwargs
+    ):
+        key = functools._make_key(args_without_self, kwargs, typed=False)
```

Comment:
it might be nice to leave a comment here explaining the risk of the memory leak?  although maybe it's fine since you wrote a test.